pkg/stanza/fileconsumer/file_test.go

```diff
@@ -403,6 +419,9 @@ func TestReadNewLogs(t *testing.T) {
 	cfg := NewConfig().includeDir(tempDir)
 	cfg.StartAt = "beginning"
 	operator, emitCalls := buildTestManagerWithOptions(t, cfg)
+	if useThreadPool.IsEnabled() {
+		operator, emitCalls = buildTestManagerWithOptions(t, cfg, withReaderChan())
+	}
```

Comment:
We could probably make better use of the options pattern here, since we are doing this everywhere. 

Maybe the options can be set in `TestMain` and then we can always just call `buildTestManager(t, cfg, options...)`